### ISSUE TITLE
Query Monitor blamed for external errors.

### ISSUE THREAD

**Issue by svandragt** [johnbillion/query-monitor#502]
It seems that Query Monitor, now it’s handling errors, is blamed by the WP Recovery Mode featurr for the errors of other plugins:

![image](https://user-images.githubusercontent.com/594871/75357896-fddc1e00-58a9-11ea-9df5-1304d647c0c4.png)

This email is sent by `class-wp-recovery-mode-email-service.php`. I'm not seeing a useful filter point here but perhaps it's possible to rig something at the point of handling the error.

**Comment by svandragt** [comment]
The full .eml report has been enclosed here in compressed form.
[WNObObX9muPP3YG35IUjFeYloCYf1h6t3LUKwrRIVkQ=@mailhog.example.eml.zip](https://github.com/johnbillion/query-monitor/files/4256476/WNObObX9muPP3YG35IUjFeYloCYf1h6t3LUKwrRIVkQ.%40mailhog.example.eml.zip)

**Comment by ihorvorotnov** [comment]
@svandragt first of all, instead of sharing a zip (many people would refuse even to download it) you can just paste the source code here. Second, the Query Monitor is not producing errors, it actually catches them - errors produced by some other code. The reason you thought it's QM is because QM Collector is the first entry in call stack:

```
Error Details
=============
An error of type E_ERROR was caused in line 82 of the file /vagrant/content/plugins/query-monitor/collectors/php_errors.php. Error message: Uncaught ParseError: syntax error, unexpected '}' in /vagrant/content/plugins/svd-cron/svd-cron.php:44
Stack trace:
#0 /vagrant/wp-config.php(145): require_once()
#1 /vagrant/wp/wp-load.php(42): require_once('/vagrant/wp-con...')
#2 /vagrant/wp/wp-admin/admin-ajax.php(22): require_once('/vagrant/wp/wp-...')
#3 {main}

Next Exception: syntax error, unexpected '}' in /vagrant/content/plugins/query-monitor/collectors/php_errors.php:82
Stack trace:
#0 [internal function]: QM_Collector_PHP_Errors->exception_handler(Object(ParseError))
#1 {main}
  thrown
```
But the actual error is here: 
```
ParseError: syntax error, unexpected '}' in /vagrant/content/plugins/svd-cron/svd-cron.php:44
```
In svd-cron plugin, main plugin file on line 44 there's a redundant `}` symbol.

**Comment by johnbillion** [comment]
Sander's point is that the email that's sent from WordPress explicitly blames QM:

> In this case, WordPress caught an error with one of your plugins, Query Monitor.

QM should be able to do something about that. Maybe. Hopefully.

**Comment by ihorvorotnov** [comment]
Ah, my bad, sorry. Didn't understand the issue and jumped right in 🤦🏻‍♂️

**Comment by crstauf** [comment]
🤞

**Comment by missikk2** [comment]
I got this email as well 4 days ago.

`Error Details
=============
An error of type E_ERROR was caused in line 82 of the file /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-content/plugins/query-monitor/collectors/php_errors.php. Error message: Uncaught Error: Call to undefined method RevSliderFunctions::get_addition() in /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-content/plugins/revslider/admin/views/modals-copyright.php:15
Stack trace:
#0 /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-content/plugins/revslider/admin/includes/shortcode_generator/shortcode_generator.class.php(242): require_once()
#1 /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-includes/class-wp-hook.php(288): RevSliderShortcodeWizard::enqueue_files('update.php')
#2 /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-includes/class-wp-hook.php(312): WP_Hook->apply_filters(NULL, Array)
#3 /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-includes/plugin.php(478): WP_Hook->do_action(Array)
#4 /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-admin/includes/template.php(2049): do_action('admin_footer', 'update.php')
#5 /nfs/c10/h11/mnt/227011/domains/arborhousefmc.com/html/wp-admin/update.php(49): iframe_footer()
#6 {main}`

**Comment by johnbillion** [comment]
This should be fixed by 355f9a6.
